jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,43 +149,173 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 
+	private String imagesDir;
+	private boolean leaveRunning;
+	private boolean shellJob;
+	private boolean extUnixSupport;
+	private int logLevel;
+	private String logFile;
+	private boolean fileLocks;
+	private String workDir;
+
+	/**
+	 * Sets the directory that will hold the images upon dump. Corresponds to criu_set_images_dir_fd.
+	 * This must be set before calling {@link #checkPointJVM()}.
+	 * 
+	 * @param imagesDir the directory as a java.nio.file.Path
+	 * @return this
+	 * @throws NullPointerException if imagesDir is null.
+	 * @throws SecurityException if no permission to access imagesDir.
+	 * @throws IllegalArgumentException if imagesDir is not a valid directory.
+	 */
+	public CRIUSupport setImagesDir(Path imagesDir) {
+		Objects.requireNonNull(imagesDir, "Path cannot be null");
+		if (!Files.isDirectory(imagesDir)) {
+			throw new IllegalArgumentException("imagesDir is not a valid directory");
+		}
+		String dir = imagesDir.toAbsolutePath().toString();
+
+		SecurityManager manager = System.getSecurityManager();
+		if (manager != null) {
+			manager.checkWrite(dir);
+		}
+		
+		this.imagesDir = dir;
+		return this;
+	}
+
+	/**
+	 * Controls whether processes are left running after dump. Corresponds to criu_set_leave_running.
```

Comment:
`criu_set_leave_running` is an implementation detail that doesn't belong here.

Reply:
Why plural? What process beside the current one might that apply to?

Reply:
I think it means child processes. https://criu.org/Final_states
I will change it to "tasks" to make it clearer.

Reply:
I wasn't aware that a snapshot applied to a 'process tree' (a phrase used in that document); using 'process tree' might be better than 'processes' (or tasks).